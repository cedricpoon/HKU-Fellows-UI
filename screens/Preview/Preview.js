import React, { Component } from 'react';
import { Container, Content } from 'native-base';

import { PostScrollable, PreviewFooter, PreviewHeader } from 'hkufui/components';
import postList from 'hkufui/docs/postList.json';

class Preview extends Component {

  render() {
    return (
      <Container>
        <PreviewHeader location={postList.location} />
        <Content>
          <PostScrollable posts={postList.posts} />
        </Content>
        <PreviewFooter />
      </Container>
    );
  }
}

export default Preview;
