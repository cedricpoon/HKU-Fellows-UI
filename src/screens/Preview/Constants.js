export const SCROLLABLE_END_REACH_THRESHOLD = 0.35;

export const ITEM_HEIGHT = 82.5; /* predefined for better performance */
