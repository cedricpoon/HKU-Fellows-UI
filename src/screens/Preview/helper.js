export function classifyQuery (query) {
  // this method should return either keyword object or solid query string
  if (query && /^#.*[^# ]$/.test(query)) {
    // keyword pattern: '#xxx#yyy'
    const keywords = query.replace(/ /g, '').split('#').filter(Boolean);
    return {
      primary: keywords[0],
      ...(keywords[1] && { secondary: keywords[1] })
    }
  } else {
    // query pattern: 'xxxyyy'
    return query;
  }
}
