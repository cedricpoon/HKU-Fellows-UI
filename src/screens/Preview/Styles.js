import { StyleSheet } from 'react-native';
import { fontSize, gridBase } from 'hkufui/theme/grid';
import { ITEM_HEIGHT } from './Constants';

export default StyleSheet.create({
  copyright: {
    marginTop: gridBase * 2,
    marginLeft: gridBase * 6,
    marginBottom: gridBase * 8,
    fontSize: fontSize.miniSize
  },
  heightHolder: {
    height: ITEM_HEIGHT
  }
});
