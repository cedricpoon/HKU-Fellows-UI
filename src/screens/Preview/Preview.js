import React, { Component } from 'react';
import { RefreshControl } from 'react-native';
import { Container, Text } from 'native-base';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { PreviewFooter, PreviewHeader, PostLoadIndicator } from 'hkufui/components';
import PostPreviewLoader from './PostPreviewLoader/PostPreviewLoader'

import * as _expandStatus from 'hkufui/src/constants/expandStatus';
import { BLAND, EXPANDING } from 'hkufui/src/constants/expandStatus';
import * as _loadStatus from 'hkufui/src/constants/loadStatus';
import { OK } from 'hkufui/src/constants/loadStatus';

import { fetchPostsSafe } from './PostPreviewLoader/loadPosts';
import { fetchExpansion } from './expandPosts';
import { onUpdateFilter, onUpdateQuery, onUpdateKeyword, resetQueryKeyword } from './filterPosts';
import styles from './Styles';
import { classifyQuery } from './helper';

import { circularTint } from 'hkufui/theme/palette';
import { localize } from 'hkufui/locale';
import { SCROLLABLE_END_REACH_THRESHOLD, ITEM_HEIGHT } from './Constants';

const locale = localize({ language: 'en', country: 'hk' });

export class Preview extends Component {
  constructor(props) {
    super(props);
    this.state = { willRefresh: false, isReady: false, searched: false };
  }

  _refresh = () => {
    this.props.onLoadPost();
  }

  _scrollInit = () => {
    this.setState({ isReady: true });
  }

  _loadMore = () => {
    if (this.state.isReady && this.props.expandStatus === BLAND) {
      this.props.onLoadMore();
    }
  }

  _scrollableUnmount = () => {
    this.setState({ isReady: false });
  }

  _renderRefreshControl = () => {
    return(
      <RefreshControl
        refreshing={false}
        onRefresh={this._refresh}
        tintColor={circularTint}
      />
    );
  }

  _renderListFooter = () => {
    const { expandStatus } = this.props;

    if (expandStatus === EXPANDING) {
      return (<PostLoadIndicator stalled />);
    }
    /* no more post footer */
    return (
      <Text note style={styles.copyright}>
        {locale['post.copyright']}
      </Text>
    );
  }

  _filterWrapper = (filter) => {
    const { onUpdateFilter } = this.props;
    return () => {
      onUpdateFilter(filter);
      this.footer.filterToggle();
    };
  }

  _searchWrapper = (query) => {
    const { onUpdateQuery, onUpdateKeyword } = this.props;
    return () => {
      const target = classifyQuery(query);
      this.setState({ searched: true });

      if (typeof target === 'object') {
        onUpdateKeyword(target);
      } else {
        onUpdateQuery(target);
      }
    }
  }

  componentDidUpdate() {
    const { shouldSearchBarEmpty } = this.props;
    const { searched } = this.state;

    if (this._header && searched && shouldSearchBarEmpty) {
      this._header.searchCancel();
    }
  }

  render() {
    const { location, loadStatus, currentFilter, onResetQueryKeyword } = this.props;
    const cancel = (isNotEmpty) => {
      if (isNotEmpty) {
        onResetQueryKeyword();
        this.setState({ searched: false });
      }
    }
    return (
      <Container>
        <PreviewHeader
          location={location}
          onSearchThunk={this._searchWrapper}
          onCancel={cancel}
          ref={ref => this._header = ref}
        />
        <PostPreviewLoader
          onScroll={this._scrollInit}
          onUnmount={this._scrollableUnmount}
          onEndReachedThreshold={SCROLLABLE_END_REACH_THRESHOLD}
          onEndReached={this._loadMore}
          refreshControl={this._renderRefreshControl()}
          ListFooterComponent={this._renderListFooter}
          removeClippedSubviews={true}
          getItemLayout={(data, index) => (
            {length: ITEM_HEIGHT, offset: ITEM_HEIGHT * index, index}
          )}
        />
        <PreviewFooter
          muted={location === ''}
          onRefresh={this._refresh}
          refreshing={loadStatus !== OK}
          popupProps={{
            onFilterThunk: this._filterWrapper,
            disabled: currentFilter
          }}
          ref={ref => this.footer = ref}
        />
      </Container>
    );
  }
}

Preview.defaultProps = {
  location: ''
}

Preview.propTypes = {
  location: PropTypes.string,
  onLoadPost: PropTypes.func.isRequired,
  onLoadMore: PropTypes.func.isRequired,
  onUpdateFilter: PropTypes.func.isRequired,
  onUpdateKeyword: PropTypes.func.isRequired,
  onUpdateQuery: PropTypes.func.isRequired,
  onResetQueryKeyword: PropTypes.func.isRequired,
  expandStatus: PropTypes.oneOf(Object.values(_expandStatus)).isRequired,
  loadStatus: PropTypes.oneOf(Object.values(_loadStatus)).isRequired,
  currentFilter: PropTypes.number.isRequired,
  shouldSearchBarEmpty: PropTypes.bool
}

const mapStateToProps = state => ({
  location: state.location.courseTitle,
  expandStatus: state.posts.subStatus,
  loadStatus: state.posts.status,
  currentFilter: state.posts.filter,
  shouldSearchBarEmpty: !state.posts.query && !state.posts.hashtag
});

const mapDispatchToProps = dispatch => ({
  onLoadPost: () => dispatch(fetchPostsSafe()),
  onLoadMore: () => dispatch(fetchExpansion()),
  onUpdateFilter: (filter) => dispatch(fetchPostsSafe(onUpdateFilter(filter))),
  onUpdateKeyword: (keyword) => dispatch(dispatch => {
    dispatch(onUpdateKeyword(keyword));
    dispatch(fetchPostsSafe());
  }),
  onUpdateQuery: (query) => dispatch(dispatch => {
    dispatch(onUpdateQuery(query));
    dispatch(fetchPostsSafe());
  }),
  onResetQueryKeyword: () => dispatch(dispatch => {
    dispatch(resetQueryKeyword());
    dispatch(fetchPostsSafe());
  })
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Preview);
