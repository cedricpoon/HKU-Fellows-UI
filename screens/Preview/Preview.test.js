import React from 'react';
import { shallow } from 'enzyme';
import Preview from './Preview';

describe('Testing Preview Screen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <Preview />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
