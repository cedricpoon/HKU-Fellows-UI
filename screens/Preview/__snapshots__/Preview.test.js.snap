// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Preview Screen renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Preview />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <withNavigation(PreviewHeader)
          location="COMP 4801"
        />,
        <Styled(Content)>
          <PostScrollable
            posts={
              Array [
                Object {
                  "id": "0",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": true,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 10,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Title",
                },
                Object {
                  "id": "1",
                  "native": false,
                  "replyNo": 12,
                  "timestamp": "2018-11-03 21:56:00",
                  "title": "Topic from Moodle",
                  "viewed": true,
                },
                Object {
                  "id": "2",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 2,
                  "solved": true,
                  "subTitle": "Proin imperdiet arcu sed velit tincidunt, eu aliquet nunc volutpat. Vivamus bibendum turpis nisi, non porta lorem aliquet aliquet.",
                  "temperature": -10,
                  "timestamp": "2018-10-22 14:00:12",
                  "title": "This is a relatively long topic that shows Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris suscipit quis sapien sed fringilla.",
                  "viewed": true,
                },
                Object {
                  "id": "3",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                  "viewed": true,
                },
                Object {
                  "id": "4",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                  "viewed": true,
                },
                Object {
                  "id": "5",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                },
                Object {
                  "id": "6",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                },
                Object {
                  "id": "7",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                },
                Object {
                  "id": "8",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                  "viewed": true,
                },
                Object {
                  "id": "9",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                },
              ]
            }
          />
        </Styled(Content)>,
        <withNavigation(PreviewFooter) />,
      ],
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "location": "COMP 4801",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <PostScrollable
            posts={
              Array [
                Object {
                  "id": "0",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": true,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 10,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Title",
                },
                Object {
                  "id": "1",
                  "native": false,
                  "replyNo": 12,
                  "timestamp": "2018-11-03 21:56:00",
                  "title": "Topic from Moodle",
                  "viewed": true,
                },
                Object {
                  "id": "2",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 2,
                  "solved": true,
                  "subTitle": "Proin imperdiet arcu sed velit tincidunt, eu aliquet nunc volutpat. Vivamus bibendum turpis nisi, non porta lorem aliquet aliquet.",
                  "temperature": -10,
                  "timestamp": "2018-10-22 14:00:12",
                  "title": "This is a relatively long topic that shows Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris suscipit quis sapien sed fringilla.",
                  "viewed": true,
                },
                Object {
                  "id": "3",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                  "viewed": true,
                },
                Object {
                  "id": "4",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                  "viewed": true,
                },
                Object {
                  "id": "5",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                },
                Object {
                  "id": "6",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                },
                Object {
                  "id": "7",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                },
                Object {
                  "id": "8",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                  "viewed": true,
                },
                Object {
                  "id": "9",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                },
              ]
            }
          />,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "posts": Array [
              Object {
                "id": "0",
                "native": true,
                "primaryHashtag": "hashtag1",
                "replyNo": 12,
                "secondaryHashtag": "hashtag2",
                "solved": true,
                "subTitle": "Subtitle to be shown",
                "temperature": 10,
                "timestamp": "2016-06-10 12:12:12",
                "title": "Title",
              },
              Object {
                "id": "1",
                "native": false,
                "replyNo": 12,
                "timestamp": "2018-11-03 21:56:00",
                "title": "Topic from Moodle",
                "viewed": true,
              },
              Object {
                "id": "2",
                "native": true,
                "primaryHashtag": "hashtag1",
                "replyNo": 2,
                "solved": true,
                "subTitle": "Proin imperdiet arcu sed velit tincidunt, eu aliquet nunc volutpat. Vivamus bibendum turpis nisi, non porta lorem aliquet aliquet.",
                "temperature": -10,
                "timestamp": "2018-10-22 14:00:12",
                "title": "This is a relatively long topic that shows Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris suscipit quis sapien sed fringilla.",
                "viewed": true,
              },
              Object {
                "id": "3",
                "native": true,
                "primaryHashtag": "hashtag1",
                "replyNo": 12,
                "secondaryHashtag": "hashtag2",
                "solved": false,
                "subTitle": "Subtitle to be shown",
                "temperature": 1,
                "timestamp": "2016-06-10 12:12:12",
                "title": "Short title",
                "viewed": true,
              },
              Object {
                "id": "4",
                "native": true,
                "primaryHashtag": "hashtag1",
                "replyNo": 12,
                "secondaryHashtag": "hashtag2",
                "solved": false,
                "subTitle": "Subtitle to be shown",
                "temperature": 1,
                "timestamp": "2016-06-10 12:12:12",
                "title": "Short title",
                "viewed": true,
              },
              Object {
                "id": "5",
                "native": true,
                "primaryHashtag": "hashtag1",
                "replyNo": 12,
                "secondaryHashtag": "hashtag2",
                "solved": false,
                "subTitle": "Subtitle to be shown",
                "temperature": 1,
                "timestamp": "2016-06-10 12:12:12",
                "title": "Short title",
              },
              Object {
                "id": "6",
                "native": true,
                "primaryHashtag": "hashtag1",
                "replyNo": 12,
                "secondaryHashtag": "hashtag2",
                "solved": false,
                "subTitle": "Subtitle to be shown",
                "temperature": 1,
                "timestamp": "2016-06-10 12:12:12",
                "title": "Short title",
              },
              Object {
                "id": "7",
                "native": true,
                "primaryHashtag": "hashtag1",
                "replyNo": 12,
                "secondaryHashtag": "hashtag2",
                "solved": false,
                "subTitle": "Subtitle to be shown",
                "temperature": 1,
                "timestamp": "2016-06-10 12:12:12",
                "title": "Short title",
              },
              Object {
                "id": "8",
                "native": true,
                "primaryHashtag": "hashtag1",
                "replyNo": 12,
                "secondaryHashtag": "hashtag2",
                "solved": false,
                "subTitle": "Subtitle to be shown",
                "temperature": 1,
                "timestamp": "2016-06-10 12:12:12",
                "title": "Short title",
                "viewed": true,
              },
              Object {
                "id": "9",
                "native": true,
                "primaryHashtag": "hashtag1",
                "replyNo": 12,
                "secondaryHashtag": "hashtag2",
                "solved": false,
                "subTitle": "Subtitle to be shown",
                "temperature": 1,
                "timestamp": "2016-06-10 12:12:12",
                "title": "Short title",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <withNavigation(PreviewHeader)
            location="COMP 4801"
          />,
          <Styled(Content)>
            <PostScrollable
              posts={
                Array [
                  Object {
                    "id": "0",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": true,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 10,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Title",
                  },
                  Object {
                    "id": "1",
                    "native": false,
                    "replyNo": 12,
                    "timestamp": "2018-11-03 21:56:00",
                    "title": "Topic from Moodle",
                    "viewed": true,
                  },
                  Object {
                    "id": "2",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 2,
                    "solved": true,
                    "subTitle": "Proin imperdiet arcu sed velit tincidunt, eu aliquet nunc volutpat. Vivamus bibendum turpis nisi, non porta lorem aliquet aliquet.",
                    "temperature": -10,
                    "timestamp": "2018-10-22 14:00:12",
                    "title": "This is a relatively long topic that shows Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris suscipit quis sapien sed fringilla.",
                    "viewed": true,
                  },
                  Object {
                    "id": "3",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                    "viewed": true,
                  },
                  Object {
                    "id": "4",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                    "viewed": true,
                  },
                  Object {
                    "id": "5",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                  },
                  Object {
                    "id": "6",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                  },
                  Object {
                    "id": "7",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                  },
                  Object {
                    "id": "8",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                    "viewed": true,
                  },
                  Object {
                    "id": "9",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                  },
                ]
              }
            />
          </Styled(Content)>,
          <withNavigation(PreviewFooter) />,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "location": "COMP 4801",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <PostScrollable
              posts={
                Array [
                  Object {
                    "id": "0",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": true,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 10,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Title",
                  },
                  Object {
                    "id": "1",
                    "native": false,
                    "replyNo": 12,
                    "timestamp": "2018-11-03 21:56:00",
                    "title": "Topic from Moodle",
                    "viewed": true,
                  },
                  Object {
                    "id": "2",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 2,
                    "solved": true,
                    "subTitle": "Proin imperdiet arcu sed velit tincidunt, eu aliquet nunc volutpat. Vivamus bibendum turpis nisi, non porta lorem aliquet aliquet.",
                    "temperature": -10,
                    "timestamp": "2018-10-22 14:00:12",
                    "title": "This is a relatively long topic that shows Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris suscipit quis sapien sed fringilla.",
                    "viewed": true,
                  },
                  Object {
                    "id": "3",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                    "viewed": true,
                  },
                  Object {
                    "id": "4",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                    "viewed": true,
                  },
                  Object {
                    "id": "5",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                  },
                  Object {
                    "id": "6",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                  },
                  Object {
                    "id": "7",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                  },
                  Object {
                    "id": "8",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                    "viewed": true,
                  },
                  Object {
                    "id": "9",
                    "native": true,
                    "primaryHashtag": "hashtag1",
                    "replyNo": 12,
                    "secondaryHashtag": "hashtag2",
                    "solved": false,
                    "subTitle": "Subtitle to be shown",
                    "temperature": 1,
                    "timestamp": "2016-06-10 12:12:12",
                    "title": "Short title",
                  },
                ]
              }
            />,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "posts": Array [
                Object {
                  "id": "0",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": true,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 10,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Title",
                },
                Object {
                  "id": "1",
                  "native": false,
                  "replyNo": 12,
                  "timestamp": "2018-11-03 21:56:00",
                  "title": "Topic from Moodle",
                  "viewed": true,
                },
                Object {
                  "id": "2",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 2,
                  "solved": true,
                  "subTitle": "Proin imperdiet arcu sed velit tincidunt, eu aliquet nunc volutpat. Vivamus bibendum turpis nisi, non porta lorem aliquet aliquet.",
                  "temperature": -10,
                  "timestamp": "2018-10-22 14:00:12",
                  "title": "This is a relatively long topic that shows Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris suscipit quis sapien sed fringilla.",
                  "viewed": true,
                },
                Object {
                  "id": "3",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                  "viewed": true,
                },
                Object {
                  "id": "4",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                  "viewed": true,
                },
                Object {
                  "id": "5",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                },
                Object {
                  "id": "6",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                },
                Object {
                  "id": "7",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                },
                Object {
                  "id": "8",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                  "viewed": true,
                },
                Object {
                  "id": "9",
                  "native": true,
                  "primaryHashtag": "hashtag1",
                  "replyNo": 12,
                  "secondaryHashtag": "hashtag2",
                  "solved": false,
                  "subTitle": "Subtitle to be shown",
                  "temperature": 1,
                  "timestamp": "2016-06-10 12:12:12",
                  "title": "Short title",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
